feature/topic/src/test/kotlin/com/google/samples/apps/nowinandroid/feature/topic/TopicViewModelTest.kt

```diff
@@ -60,7 +59,8 @@ class TopicViewModelTest {
     @Before
     fun setup() {
         viewModel = TopicViewModel(
-            savedStateHandle = SavedStateHandle(mapOf(TOPIC_ID_ARG to testInputTopics[0].topic.id)),
+            // TODO: Figure out how to supply the correct dependency TopicDestination(id = testInputTopics[0].topic.id)
```

Comment:
This is ViewModel Unit test, so that I don't think here is the place to consider Navigation Args.
Manual insert `SavedStateHandle` looks fine to me.

Reply:
I think what he meant was to be able to not rely on the `TOPIC_ID_ARG` hardcoded parameter, and instead being able to generate a `SavedStateHandle` from a `TopicDestination` instance.
There is this extension that takes a `SavedStateHandle` and returns your Destination data class:

```kotlin
public inline fun <reified T> SavedStateHandle.toRoute(): T = serializer<T>().decodeArguments(this)
```

There should probably be a test fixture to do the opposite. The closest I found is in `RouteSerializerKt#generateRouteWithArgs()` but that does not solves the issue entirely.

I suppose the key is based on the Serializable, and therefore could be accessed in tests with `TopicDestination::topicId.name`

Reply:
Is there any update to the new API? 
```kotlin
public operator fun SavedStateHandle.Companion.invoke(
    route: Any,
    typeMap: Map<KType, @JvmSuppressWildcards NavType<*>> = emptyMap()
): SavedStateHandle
```

Hopefully, they'll include a documentation for its usage.

Reply:
The issue of `SavedStateHandle.toRoute` causing ViewModel tests to be instrumented (i.e. needing an emulator or Robolectric) is tracked here: https://issuetracker.google.com/349807172. Feel free to upvote if this affects you.